### PR TITLE
added home work 33

### PR DESCRIPTION
**Pull request by NataliiaFed**
Home work 33
I was thinking of using classes for books/<book_title>, users/<user_id and params views, but haven't figured out how to pass route variables into add_url_rule method.
Also was thinking of applying a decorator to all views in task 4, to perform validation. But also haven't figured out if is implementable

### CODE REVIEW

File: flask-homework/my_app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    'version': 1,
+    'formatters': {'default': {
+        'format': '[%(asctime)s] %(levelname)s in %(module)s: %(message)s'
+    }},
+})
+
+app.config.from_object(AppConfig)
+
+from .views import *
+from .class_base_views import *
```
Comment (klymenok) [inline flask-homework/my_app/__init__.py]: There is no such file `class_base_views`

**Comment by klymenok** [comment]
> 

Here is an example of class-based view with parameter:
```

class MyDetailView(View):
    methods = ['GET']

    def dispatch_request(self, id):
        return f'Id is {id}', 200


app.add_url_rule(
    '/class/users/<id>',
    view_func=MyDetailView.as_view('class-user-detail')
)
```

**Comment by klymenok** [comment]
> Home work 33 I was thinking of using classes for books/<book_title>, users/<user_id and params views, but haven't figured out how to pass route variables into add_url_rule method. Also was thinking of applying a decorator to all views in task 4, to perform validation. But also haven't figured out if is implementable

Here is an example of decorator to check if user is logged in:
```
def is_logged_in(func):
    def wrap(*args, **kwargs):
        if not session.get('username'):
            return redirect(url_for('login'))
        return func(*args, **kwargs)
    return wrap

@app.route('/users')
@is_logged_in
def users():
    query = db.select(User)
    users = db.session.scalars(query)
    return render_template('users.html', users=users)

```

**Review by klymenok — CHANGES_REQUESTED** [review]

File: flask-homework/my_app/__init__.py (hunk 1 above)
Comment (NataliiaFed) [reply flask-homework/my_app/__init__.py]: Yeah, write! That was me trying to implement class-based viewes. And I forgot to delete this line after unsuccessful attempts. Will try again using your hints within the next homework

**Review by klymenok — APPROVED** [review]
Looks good1

**Pull request merged**